pandas/util/depr_module.py

```diff
@@ -0,0 +1,57 @@
+"""
+This module houses a utility class for mocking deprecated modules.
+It is for internal use only and should not be used beyond this purpose.
+"""
+
+import warnings
+import importlib
+
+
+class _DeprecatedModule(object):
+    """ Class for mocking deprecated modules.
+
+    Parameters
+    ----------
+    deprmod : name of module to be deprecated.
+    alts : alternative modules to be used to access objects or methods
+           available in module.
+    removals : objects or methods in module that will no longer be
+               accessible once module is removed.
+    """
+    def __init__(self, deprmod, alts=None, removals=None):
+        self.deprmod = deprmod
+        self.alts = alts
+        self.removals = removals
+
+    def __getattr__(self, name):
+        if name in dir(self.__class__):
```

Comment:
set this on creation as this is a list so checking will be expensive (make it a set)